rkt/images.go

```diff
@@ -224,49 +222,93 @@ func (f *fetcher) fetchSingleImage(img string, asc string, discover bool) (strin
 		return "", fmt.Errorf("unable to access %q: %v", img, err)
 	}
 
-	if discover && u.Scheme == "" {
-		if app := newDiscoveryApp(img); app != nil {
-			var discoveryError error
-			if !f.local {
-				stderr("rkt: searching for app image %s", img)
-				ep, err := discoverApp(app, true)
-				if err != nil {
-					discoveryError = err
-				} else {
-					// No specified version label, mark it as latest
-					if _, ok := app.Labels["version"]; !ok {
-						latest = true
-					}
-					return f.fetchImageFromEndpoints(app.Name.String(), ep, ascFile, latest)
-				}
-			}
-			if discoveryError != nil {
-				stderr("discovery failed for %q: %v. Trying to find image in the store.", img, discoveryError)
-			}
-			if f.local || discoveryError != nil {
-				return f.fetchImageFromStore(img)
-			}
+	switch u.Scheme {
+	// image name
+	case "":
+		return f.fetchImageByName(img, ascFile)
+	case "http", "https", "file", "docker":
+		return f.fetchImageByURL(u, ascFile)
+	}
+
+	return "", fmt.Errorf("rkt only supports fetching for image name and http, https, docker or file URLs (%s)", img)
+}
+
+func (f *fetcher) fetchImageByName(img string, ascFile *os.File) (string, error) {
```

Comment:
> to check my understanding, since I think we overload terms a little bit: the "Name" == as defined here?

Right. It's the "image name string"

> we should hurry up and standardise this since it's confusing to refer to it elsewhere as e.g. getStoreKeyFrom*App*

Definitely. :smile: 

> mind adding a docstring here?

done.